### PR TITLE
Reduce potential triggers for building code on build servers (Linux/macOS)

### PR DESCRIPTION
**Pull request by martyngigg**
Description of work.

As we move the developer documentation into the repository it is becoming more likely that pull requests will be submitted with purely documentation changes . This PR adds further checks so the code + unit tests is only built when necessary.

**To test:**

* The builds should pass
* Review the new logic thoroughly (twice to be sure!)

*No issue*

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: buildconfig/Jenkins/check_for_changes [hunk 1]
```diff
@@ -46,6 +46,26 @@ case "$TYPE" in
 	    exit $NOTFOUND
 	fi
     ;;
+    dev-docs-only)
+	# FOUND=1 iff changes are limited to dev-docs only
+	# Find all changed files and grep for required type. -v inverts match so grep=0 means
+	# there are other changes besides this
+	if git diff --name-only ${BRANCH_TIP} ${BRANCH_BASE} -- | grep -q -E -v '^dev-docs/'; then
```
Comment (peterfpeterson) [inline buildconfig/Jenkins/check_for_changes]: Why not
```shell
if $(! git diff --quiet ${BRANCH_TIP} ${BRANCH_BASE} -- dev-docs)
then
  exit $FOUND
fi
exit $NOTFOUND
```
Someone more thoughtful could probably make it a one-liner.

**Review by peterfpeterson — COMMENTED** [review]
I *think* this does the right thing.

File: buildconfig/Jenkins/check_for_changes (hunk 1 above)
Comment (peterfpeterson) [reply buildconfig/Jenkins/check_for_changes]: Even shorter option (for the lines above too):
```shell
git diff --quiet ${BRANCH_TIP} ${BRANCH_BASE} -- dev-docs || return $FOUND
return $NOTFOUND
```
Then the things earlier could be single lines (for finding) that ends in `$NOTFOUND`. It's clearer to me, but maybe not others.

Comment (martyngigg) [reply buildconfig/Jenkins/check_for_changes]: Thanks for these. The script looks a little cleaner now.

**Review by peterfpeterson — APPROVED** [review]
LGTM

**Pull request merged**